File: BACKEND/ValoracionService/src/main/java/dev/jvall/ValoracionService/domain/Valoracion.java
package dev.jvall.ValoracionService.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import reactor.util.annotation.Nullable;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Valoracion {
    private Integer idvaloracion;
    private String usuario;
    private String nombres;
    private Integer valoracion;
    private String opinion;
    private String fecha;
    @Nullable
    private String clasificacion;
    private Integer idpedido;
}

File: FRONTEND/sm-web/src/app/backend/interfaces/valoracion.ts
export interface Valoracion{
    idvaloracion:number,
    usuario:string,
    nombres:string,
    valoracion:number,
    opinion:string,
    fecha:string,
    clasificacion?:string,
    idpedido:number
}
export interface ValoracionRequest{
    usuario:string,
    nombres:string,
    valoracion:number,
    opinion: string;
    clasificacion:string,
    idpedidoin:number
}